geniza/corpus/management/commands/sync_transcriptions.py

```diff
@@ -0,0 +1,151 @@
+        for xmlfile in glob.iglob(os.path.join(gitrepo_path, "*.xml")):
+            stats["xml"] += 1
+
+            tei = xmlmap.load_xmlobject_from_file(xmlfile, GenizaTei)
+            # some files are stubs with no content
+            # check if there is no text content; report and skip
+            if tei.no_content():
+                self.stdout.write("%s has no text content, skipping" % xmlfile)
+                stats["empty_tei"] += 1
+                continue
+
+            # get the document id from the filename (####.xml)
+            pgpid = os.path.splitext(os.path.basename(xmlfile))[0]
+            # in ONE case there is a duplicate id with b suffix on the second
+            try:
+                pgpid = int(pgpid.strip("b"))
+            except ValueError:
+                self.stderr.write("Failed to generate integer PGPID from %s" % pgpid)
+                continue
+
+            # find the document in the database
+            try:
+                doc = Document.objects.get(
+                    models.Q(id=pgpid) | models.Q(old_pgpids__contains=[pgpid])
+                )
+            except Document.DoesNotExist:
+                stats["document_not_found"] += 1
+                self.stdout.write("Document %s not found in database" % pgpid)
+                continue
+
+            if doc.fragments.count() > 1:
+                stats["joins"] += 1
+
+            footnote = None
+
+            editions = doc.footnotes.editions()
+            if editions.count() > 1:
+                # debugging output for footnote selection
+                # print('more than one edition for %s' % xmlfile)
+                # print(list(ed.source for ed in editions))
+                # try filtering by current text content
+                editions_with_content = editions.filter(content__isnull=False)
+                # print('editions with content')
+                # print(list(ed.source for ed in editions_with_content))
+                stats["multiple_editions"] += 1
+                if editions_with_content.count() > 1:
+                    stats["multiple_editions_with_content"] += 1
+                    multiedition_docs.add(doc.id)
+                elif editions_with_content.count() == 1:
+                    # if there was only one, assume it's the one to update
+                    footnote = editions_with_content.first()
+            elif not editions.exists():
+                # debugging output for footnote selection
+                # print('no edition for %s' % xmlfile)
+                stats["no_edition"] += 1
+            else:
+                stats["one_edition"] += 1
+                # if only one edition, update the transciption content there
+                footnote = editions.first()
+
+            # if we identified the correct footnote, update it
+            if footnote:
+                html = tei.text_to_html()
+                if html:
+                    footnote.content = {"html": html}
```

Comment:
In my local, it looks like the footnotes' content is always a dict with the `lines` key, which is an array of lines. Is the idea that this should be overwritten with the new HTML, or should that other representation be preserved?

Reply:
The lines/blocks json data was the transcription we intitially populated when we imported the spreadsheet data. My plan is to replace that one with the html version — the html is still an interim solution, but it adds line numbers which are a requirement and were not included in the initial lines/blocks implementation.

FWIW, I started working on indexing and highlighting transcription text and I think I'm going to populate a plain text version of the transcription content at the same time as the html version — seems easier to do it here and combine the logic than doing it at index time.  I'll add that in a subsequent PR, though.